internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -664,7 +674,31 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *ai
 
 		return nil
 	}
-	return deployment, mutateFn
+	return deployment, mutateFn, nil
+}
+
+func getOpenshiftVersions(instance *aiv1beta1.AgentServiceConfig) (string, error) {
+	if instance.Spec.OSImages == nil {
+		return OpenshiftVersions(), nil
+	}
+
+	var versions models.OpenshiftVersions
+	for _, image := range instance.Spec.OSImages {
+		version := models.OpenshiftVersion{
+			RhcosImage: &image.Url,
```

Comment:
Do we need to check if this link is accessible?

Reply:
That's a reasonable enhancement to consider in the future.